### PR TITLE
Use rapids_cpm_bs_thread_pool()

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: rapids_config.cmake [hunk 1]
```diff
@@ -29,7 +29,7 @@ endif()
 if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/CUDF_RAPIDS-${RAPIDS_VERSION_MAJOR_MINOR}.cmake")
   file(
     DOWNLOAD
-    "https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-${RAPIDS_VERSION_MAJOR_MINOR}/RAPIDS.cmake"
+    "https://raw.githubusercontent.com/KyleFromNVIDIA/rapids-cmake/bs-thread-pool/RAPIDS.cmake"
```
Comment (KyleFromNVIDIA) [inline rapids_config.cmake:32]: This is a temporary change to test https://github.com/rapidsai/rapids-cmake/pull/651.

File: cpp/cmake/thirdparty/get_thread_pool.cmake [hunk 2]
```diff
@@ -12,20 +12,14 @@
...
-    BS_thread_pool 4.1.0
-    CPM_ARGS
-    GIT_REPOSITORY https://github.com/bshoshany/thread-pool.git
-    GIT_TAG 097aa718f25d44315cadb80b407144ad455ee4f9
-    GIT_SHALLOW TRUE
-  )
-  if(NOT TARGET BS_thread_pool)
-    add_library(BS_thread_pool INTERFACE)
-    target_include_directories(BS_thread_pool INTERFACE ${BS_thread_pool_SOURCE_DIR}/include)
-    target_compile_definitions(BS_thread_pool INTERFACE "BS_THREAD_POOL_ENABLE_PAUSE=1")
```
Comment (madsbk) [inline cpp/cmake/thirdparty/get_thread_pool.cmake:27]: Notice, `BS::thread_pool` needs to be compiled with `BS_THREAD_POOL_ENABLE_PAUSE=1` defined.

Comment (robertmaynard) [reply cpp/cmake/thirdparty/get_thread_pool.cmake:27]: That is being done unconditionally on the rapids-cmake side ( https://github.com/rapidsai/rapids-cmake/pull/651/commits/59b000ab1c1bd6b9efcd9566cacb306aa0c6227e )

**Review by bdice — APPROVED** [review]

**Pull request merged**